### ISSUE TITLE
Inconsistency of weighting internal unit

### ISSUE THREAD

**Issue by BrianMarre** [ComputationalRadiationPhysics/picongpu#3372]
The weight of a macro-particle as returned by particle[ weight_  ] is currently not in units of TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE, but rather equal to the number of physical particles it corresponds to.

- This is inconsistent with other particle attributes, which are scaled in internal units, realtive to a user configured base line.

- This might pose a numerical problem if weights reach very large values, since floating point numbers are inherently less accurate at higher values.

It would in my opinion be more consistent to also define a UNIT_WEIGHTING such that a macro particle of weighting 1 corresponds to  TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE physical particles

**Comment by PrometheusPi** [comment]
I agree with @BrianMarre, this conversion would be more consistent - but I think the naming of the conversion factor `unitSI` for the openPMD output would be misleading, since you do not convert to SI units.

**Comment by sbastrakov** [comment]
I agree with @PrometheusPi . Perhaps we could do `UNIT_WEIGHTING` as suggested, but just for getting the unit internally use the PIConGPU internal naming scheme  `getUnit()` when needed. And avoid mentioning SI as strictly speaking it's not a unit, but the weighting is unitless and conversion is just a scale.
